sdks/python/apache_beam/io/gcp/bigquery_test.py

```diff
@@ -508,21 +508,26 @@ def test_temp_dataset_is_configurable(
   ])
   def test_create_temp_dataset_exception(self, exception_type, error_message):
 
-    with mock.patch.object(bigquery_v2_client.BigqueryV2.JobsService,
+    with mock.patch.object(beam.io.gcp.bigquery._CustomBigQuerySource,
+                           'estimate_size') as mock_estimate,\
+            mock.patch.object(bigquery_v2_client.BigqueryV2.JobsService,
                            'Insert'),\
       mock.patch.object(BigQueryWrapper,
                         'get_or_create_dataset') as mock_insert, \
       mock.patch('time.sleep'), \
       self.assertRaises(Exception) as exc,\
       beam.Pipeline() as p:
 
+      mock_estimate.return_value = None
       mock_insert.side_effect = exception_type(error_message)
 
       _ = p | ReadFromBigQuery(
           project='apache-beam-testing',
           query='SELECT * FROM `project.dataset.table`',
           gcs_location='gs://temp_location')
 
+    p.run().wait_until_finish(duration=30)
```

Comment:
these will cause the pipeline to be ran twice (see https://stackoverflow.com/a/52282001/1255356)

Reply:
That is likely causing my mocking problem. What is strange then is why the exception that is getting mocked here isn't getting thrown